llvm/lib/CodeGen/MachineLateInstrsCleanup.cpp

```diff
@@ -186,9 +197,67 @@ static bool isCandidate(const MachineInstr *MI, Register &DefedReg,
     const MachineOperand &MO = MI->getOperand(i);
     if (MO.isReg()) {
       if (MO.isDef()) {
+        // To get the value for DefedReg, we need to check that 1st
+        // MachineOperand is not dead and not implicit def.
+        //
+        // For example:
+        // renamable $r9d = MOV32r0 implicit-def dead $eflags, implicit-def $r9
+        // First operand is $r9d and it is not implicit def and not dead, So
+        // it is valid and we can use it in DefedReg.
         if (i == 0 && !MO.isImplicit() && !MO.isDead())
           DefedReg = MO.getReg();
-        else
+        // If DefedReg has a valid register, check the other operands
+        else if (DefedReg != MCRegister::NoRegister) {
+          // If the machineOperand is Dead and Implicit then continue
+          // to next operand.
```

Comment:
I know I asked for comments, but I wanted something that explains _why_ the code does what it does. In this case, why do we ignore dead implicit operands, specifically?

Please remove comments like this that just state _what_ the code does. They are not helpful in any way.